src/glmtools.jl

```diff
@@ -14,8 +14,12 @@ linkinv(::CauchitLink, η) = oftype(η, 0.5) + atan(η) / pi
 mueta(::CauchitLink, η) = one(η) / (pi * (one(η) + abs2(η)))
 
 linkfun(::CloglogLink, μ) = log(-log1p(-μ))
-linkinv(::CloglogLink, η) = -expm1(-exp(η))
-mueta(::CloglogLink, η) = exp(η) * exp(-exp(η))
+function linkinv{T<:Real}(::CloglogLink, η::T)
+    clamp(-expm1(-exp(η)), eps(T), one(T) - eps(T))
```

Comment:
I know but this a compromise to make autodiff types go through and promotion shouldn't happen here anyway. We don't want an unclamped code path. It would be very tricky to figure out why code taking that path would fail.